### PR TITLE
ShapeDecoration

### PR DESCRIPTION
**Pull request by Hixie**
cc @gspencergoog

### CODE REVIEW

**Review by gspencergoog — APPROVED** [review]

**Comment by gspencergoog** [comment]
LGTM

File: packages/flutter/lib/src/painting/shape_decoration.dart [hunk 1]
```diff
@@ -0,0 +1,388 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
```
[the remark this answers is not in the document]
Comment (Hixie) [reply packages/flutter/lib/src/painting/shape_decoration.dart]: It was derived from a file with the previous copyright.

File: packages/flutter/lib/src/painting/shape_decoration.dart [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+  }) : assert(!(color != null && gradient != null)),
+       assert(shape != null);
+
+  /// Creates a shape decoration configured to match a [BoxDecoration].
+  ///
+  /// The [BoxDecoration] class is more efficient for shapes that it can
+  /// describe than the [ShapeDecoration] class is for those same shapes,
+  /// because [ShapeDecoration] has to be more general since it can support any
+  /// shape. However, having a [ShapeDecoration] is sometimes necessary, for
+  /// example when calling [ShapeDecoration.lerp].
```
[the remark this answers is not in the document]
Comment (Hixie) [reply packages/flutter/lib/src/painting/shape_decoration.dart]: added more text

**Comment by Hixie** [comment]
Will land on green.

**Pull request merged**